### PR TITLE
Multiple Preview Fixes

### PR DESCRIPTION
**Pull request by kevinslin**
Adds the following missing features to the preview

- better block quote support
- add mermaid support
- add base css 
- support katex rendering
- respect fmTitle heading

### CODE REVIEW

**Review by jonathanyeung — APPROVED** [review]
Changes looking good!  I didn't scrutinize the .scss files too closely though.

File: packages/dendron-next-server/styles/scss/content.scss [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+  cite {
+    font-style: italic;
+
+    &:before {
+      content: "\2014";
+      padding-right: 5px;
+    }
+  }
+}
```
Comment (namjul) [inline packages/dendron-next-server/styles/scss/content.scss:215]: Many hardwired style tokens in this file. I am guessing that is ok for now and at some point we want to have a more coherent system across at least the markdown preview and the published site. And maybe but I think less important also unified with the components that make up the vscode UI.

Comment (namjul) [reply packages/dendron-next-server/styles/scss/content.scss:215]: Maybe a good starting place would be to have a single source for style tokens by partially using https://github.com/system-ui/theme-specification inside `_variables.scss `

File: packages/dendron-next-server/styles/scss/support/_variables.scss [hunk 2]
```diff
@@ -0,0 +1,135 @@
+//
```
Comment (namjul) [inline packages/dendron-next-server/styles/scss/support/_variables.scss:1]: I am confused about this second `variables.scss` file. What does the `support` folder imply?

File: packages/dendron-next-server/styles/scss/content.scss (hunk 1 above)
Comment (namjul) [reply packages/dendron-next-server/styles/scss/content.scss:215]: I am just seing that we are doing this already in this PR. Keeping it for record.

File: packages/dendron-next-server/styles/scss/content.scss [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  h3,
+  h4,
+  h5,
+  h6 {
+    position: relative;
+    margin-top: 1.5em;
+    margin-bottom: 0.25em;
+
+    &:first-child {
+      margin-top: $sp-2;
```
Comment (namjul) [inline packages/dendron-next-server/styles/scss/content.scss:182]: Some places we use tokens other not.

File: packages/dendron-next-server/styles/scss/support/_variables.scss [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  sp-1: $spacing-unit * 0.25,
+  sp-2: $spacing-unit * 0.5,
+  sp-3: $spacing-unit * 0.75,
+  sp-4: $spacing-unit,
+  sp-5: $spacing-unit * 1.5,
+  sp-6: $spacing-unit * 2,
+  sp-7: $spacing-unit * 2.5,
+  sp-8: $spacing-unit * 3,
+  sp-9: $spacing-unit * 3.5,
+  sp-10: $spacing-unit * 4,
```
Comment (namjul) [inline packages/dendron-next-server/styles/scss/support/_variables.scss:87]: I like the values choosen.

**Review by namjul — COMMENTED** [review]
I am wondering what the intent is with how this PR adds new css. It has its own flavor or does it reference an existing standard (eg tailwind)? Are there rules to follow when adding css? It looks to me we sattle on scss for the markdown-preview and publishing and antd for UI components.

File: packages/dendron-next-server/styles/scss/content.scss (hunk 1 above)
Comment (kevinslin) [reply packages/dendron-next-server/styles/scss/content.scss:215]: our css is... a mess. it is currently a frankenstein of the following two projects:
- https://github.com/pmarsceill/just-the-docs
- https://github.com/mmistakes/minimal-mistakes

the css you see in this review is what was selectively migrated from [dendron-11ty](https://wiki.dendron.so/notes/rIUY5NLx0inCdmRLJOFaq.html), our current publishing template. 
if you at some point want to do a refactoring/unification of our styling, all I can say is yes, please!

Comment (namjul) [reply packages/dendron-next-server/styles/scss/content.scss:215]: Yes, I would enjoy that. Maybe a good next step would be to create an RFC.

Comment (kevinslin) [reply packages/dendron-next-server/styles/scss/content.scss:215]: that would be great. will leave it to then for the rfc :)

**Pull request closed without merging**